src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -3,157 +3,256 @@
+    }
+
+    private final Node processInitialLabel(String initialLabel, List<String> removedLabels, 
+                                            Optional<String> suggestion) {
+        TurboLabel repoInitialLabel = TurboLabel.getFirstMatchingTurboLabel(allLabels, initialLabel);
+        if (!removedLabels.contains(initialLabel)) {
+            if (suggestion.isPresent() && initialLabel.equals(suggestion.get())) {
+                return getStyledPickerLabel(
+                    new PickerLabel(repoInitialLabel, true).faded(true).removed(true));
             }
-        });
+            return getStyledPickerLabel(new PickerLabel(repoInitialLabel, true));
+        }
+
+        if (suggestion.isPresent() && initialLabel.equals(suggestion.get())) {
+            return getStyledPickerLabel(new PickerLabel(repoInitialLabel, true).faded(true));
+        }
+
+        return getStyledPickerLabel(new PickerLabel(repoInitialLabel, true).removed(true));
     }
 
-    @SuppressWarnings("unused")
-    private void ______POPULATION______() {}
+    /**
+     * @param label
+     * @return Node from label after registering mouse handler
+     */
+    private final Node getStyledPickerLabel(PickerLabel label) {
+        Node node = label.getNode();
+        node.setOnMouseClicked(e -> handleLabelClick(label));
+        return node;
+    }
 
-    protected void populatePanes(List<PickerLabel> existingLabels, List<PickerLabel> newTopLabels,
-            List<PickerLabel> bottomLabels, Map<String, Boolean> groups) {
-        populateTopPane(existingLabels, newTopLabels);
-        populateBottomBox(bottomLabels, groups);
+    private final void populateToBeAddedLabels(List<String> addedLabels, Optional<String> suggestion) {
+        if (!addedLabels.isEmpty() || hasNewSuggestion(addedLabels, suggestion)) {
+            assignedLabels.getChildren().add(new Label("|"));
+        }
+        populateAddedLabels(addedLabels, suggestion);
+        populateSuggestedLabel(addedLabels, suggestion);
     }
 
-    private void populateTopPane(List<PickerLabel> existingLabels, List<PickerLabel> newTopLabels) {
-        topPane.getChildren().clear();
-        if (existingLabels.isEmpty() && newTopLabels.isEmpty()) {
-            Label label = new Label("No currently selected labels. ");
-            label.setPadding(new Insets(2, 5, 2, 5));
-            topPane.getChildren().add(label);
-        } else {
-            existingLabels.forEach(label -> topPane.getChildren().add(label.getNode()));
-            if (!newTopLabels.isEmpty()) {
-                topPane.getChildren().add(new Label("|"));
-                newTopLabels.forEach(label -> topPane.getChildren().add(label.getNode()));
-            }
+    private final void populateAddedLabels(List<String> addedLabels, Optional<String> suggestion) {
+        addedLabels.stream()
+                .forEach(label -> {
+                    assignedLabels.getChildren().add(processAddedLabel(label, suggestion));
+                });
```

Comment:
Actually, braces is usually compulsory even for one liners, but we make an exception for cases such as guard conditions. e.g. `if (param == null) return;`